File: ArrayKata/javascript/ArrayKata.js

       let myArray = [310,25,29,23,10,14,2,110,300,4,49,2];
       console.log(maximumIn(myArray));
       console.log(minimumIn(myArray));
       console.log(sumOf(myArray));
       console.log(sumOfEven(myArray));
       console.log(sumOfOdd(myArray));
       console.log(noOfOddNumbersIn(myArray));
       console.log(noOfEvenNumbersIn(myArray));
       console.log(maximumAndMinimumOf(myArray));
       console.log(evenNumbersIn(myArray));
       console.log(oddNumbersIn(myArray));
       console.log(squareNumbersIn(myArray));

    function maximumIn(arrayOfIntegers){
        let largest = arrayOfIntegers[0];
        for(let count=0;count<arrayOfIntegers.length; count++){
            if(largest<arrayOfIntegers[count]){
                largest = arrayOfIntegers[count];
                }
            }
        return largest;
    }
    
     function minimumIn(arrayOfIntegers){
        let smallest = arrayOfIntegers[0];
        for(let count=0;count<arrayOfIntegers.length; count++){
            if(smallest>arrayOfIntegers[count]){
                smallest = arrayOfIntegers[count];
                }
            }
        return smallest;
    }

    function sumOf(arrayOfIntegers){
        let sum = 0;
        for(let count=0;count<arrayOfIntegers.length; count++){
            sum+=arrayOfIntegers[count];
            }
        return sum;
    }

     function sumOfEven(arrayOfIntegers){
        let sumOfEvenNumbers=0;
        for(let count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 == 0){
                sumOfEvenNumbers+=arrayOfIntegers[count];
                }
            }
        return sumOfEvenNumbers;
    }


     function sumOfOdd(arrayOfIntegers){
        let sumOfOddNumbers=0;
        for(let count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 != 0){
                sumOfOddNumbers+=arrayOfIntegers[count];
                }
            }
        return sumOfOddNumbers;
    }
    function maximumAndMinimumOf(arrayOfIntegers){
        let smallest = arrayOfIntegers[0];
        let largest = arrayOfIntegers[0];
        let arrayOfMinAndMax = [];
        for(let count=0;count<arrayOfIntegers.length; count++){
            if(smallest>arrayOfIntegers[count]){
                smallest = arrayOfIntegers[count];
                }
            if(largest<arrayOfIntegers[count]){
                largest = arrayOfIntegers[count];
                }
            }
        arrayOfMinAndMax[0]=smallest;
        arrayOfMinAndMax[1]=largest;
        return arrayOfMinAndMax;
    }

     function noOfOddNumbersIn(arrayOfIntegers){
        let countOfOddNumbers=0;
        for(let count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 != 0){
                countOfOddNumbers++;
                }
            }
        return countOfOddNumbers;
    }


     function noOfEvenNumbersIn(arrayOfIntegers){
        let countOfEvenNumbers=0;
        for(let count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 == 0){
                countOfEvenNumbers++;
                }
            }
        return countOfEvenNumbers;
    }

    function evenNumbersIn(arraysOfIntegers){
        let evenNumbers = []
        for(let integer of arraysOfIntegers){
            if(integer % 2 == 0){            
           evenNumbers.push(integer);
            }
        }
        return evenNumbers;
    }

   function oddNumbersIn(arraysOfIntegers){
        let oddNumbers= [];
        for(let integer of arraysOfIntegers){
            if(integer % 2 != 0){            
           oddNumbers.push(integer);
            }
        }
        return oddNumbers;
    }

    function squareNumbersIn(arrayOfIntegers){
        let squareNumbers= []
        for(let integer of arrayOfIntegers){
            let result = Math.sqrt(integer);
            if(result*10%10 == 0){
                squareNumbers.push(integer);
            }
        }
        return squareNumbers;
    }

File: ArrayKata/java/ArrayKata.java
import java.util.ArrayList;
import java.util.Arrays;
public class ArrayKata{
    public static void main(String[] args){
        int[] myArray = {310,25,29,23,10,14,2,110,300,4,49,2};
       System.out.println(maximumIn(myArray));
       System.out.println(minimumIn(myArray));
       System.out.println(sumOf(myArray));
       System.out.println(sumOfEven(myArray));
       System.out.println(sumOfOdd(myArray));
       System.out.println(noOfOddNumbersIn(myArray));
       System.out.println(noOfEvenNumbersIn(myArray));
       System.out.println(Arrays.toString(maximumAndMinimumOf(myArray)));
       System.out.println(evenNumbersIn(myArray));
       System.out.println(oddNumbersIn(myArray));
       System.out.println(squareNumbersIn(myArray));
    }

    public static int maximumIn(int[] arrayOfIntegers){
        int largest = arrayOfIntegers[0];
        for(int count=0;count<arrayOfIntegers.length; count++){
            if(largest<arrayOfIntegers[count]){
                largest = arrayOfIntegers[count];
                }
            }
        return largest;
    }
    
     public static int minimumIn(int[] arrayOfIntegers){
        int smallest = arrayOfIntegers[0];
        for(int count=0;count<arrayOfIntegers.length; count++){
            if(smallest>arrayOfIntegers[count]){
                smallest = arrayOfIntegers[count];
                }
            }
        return smallest;
    }

    public static int sumOf(int[] arrayOfIntegers){
        int sum = 0;
        for(int count=0;count<arrayOfIntegers.length; count++){
            sum+=arrayOfIntegers[count];
            }
        return sum;
    }

     public static int sumOfEven(int[] arrayOfIntegers){
        int sumOfEvenNumbers=0;
        for(int count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 == 0){
                sumOfEvenNumbers+=arrayOfIntegers[count];
                }
            }
        return sumOfEvenNumbers;
    }


     public static int sumOfOdd(int[] arrayOfIntegers){
        int sumOfOddNumbers=0;
        for(int count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 != 0){
                sumOfOddNumbers+=arrayOfIntegers[count];
                }
            }
        return sumOfOddNumbers;
    }
    public static int[] maximumAndMinimumOf(int[] arrayOfIntegers){
        int smallest = arrayOfIntegers[0];
        int largest = arrayOfIntegers[0];
        int[] arrayOfMinAndMax = new int[2];
        for(int count=0;count<arrayOfIntegers.length; count++){
            if(smallest>arrayOfIntegers[count]){
                smallest = arrayOfIntegers[count];
                }
            if(largest<arrayOfIntegers[count]){
                largest = arrayOfIntegers[count];
                }
            }
        arrayOfMinAndMax[0]=smallest;
        arrayOfMinAndMax[1]=largest;
        return arrayOfMinAndMax;
    }

     public static int noOfOddNumbersIn(int[] arrayOfIntegers){
        int countOfOddNumbers=0;
        for(int count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 != 0){
                countOfOddNumbers++;
                }
            }
        return countOfOddNumbers;
    }


     public static int noOfEvenNumbersIn(int[] arrayOfIntegers){
        int countOfEvenNumbers=0;
        for(int count=0;count<arrayOfIntegers.length; count++){
            if(arrayOfIntegers[count]%2 == 0){
                countOfEvenNumbers++;
                }
            }
        return countOfEvenNumbers;
    }

    public static ArrayList<Integer> evenNumbersIn(int[] arraysOfIntegers){
        ArrayList<Integer> evenNumbers= new ArrayList<>();
        for(int integer : arraysOfIntegers){
            if(integer % 2 == 0){            
           evenNumbers.add(integer);
            }
        }
        return evenNumbers;
    }

   public static ArrayList<Integer> oddNumbersIn(int[] arraysOfIntegers){
        ArrayList<Integer> oddNumbers= new ArrayList<>();
        for(int integer : arraysOfIntegers){
            if(integer % 2 != 0){            
           oddNumbers.add(integer);
            }
        }
        return oddNumbers;
    }

    public static ArrayList<Integer> squareNumbersIn(int[] arrayOfIntegers){
        ArrayList<Integer> squareNumbers= new ArrayList<>();
        for(int integer:arrayOfIntegers){
            double result = Math.sqrt(integer);
            if(result*10%10 == 0){
                squareNumbers.add(integer);
            }
        }
        return squareNumbers;
    }

}
